webapp/src/main/java/com/box/l10n/mojito/service/security/user/UserService.java

```diff
@@ -40,6 +46,23 @@ public class UserService {
 
   @Autowired AuditorAwareImpl auditorAwareImpl;
 
+  @Autowired(required = false)
+  ServiceDisambiguator serviceDisambiguator;
+
+  @Autowired HeaderSecurityConfig headerSecurityConfig;
+
+  PagerDutyClient pagerDutyClient;
+
+  @Autowired
+  public UserService(PagerDutyIntegrationService pagerDutyIntegrationService) {
+    Optional<PagerDutyClient> defaultPagerDutyClient =
+        pagerDutyIntegrationService.getDefaultPagerDutyClient();
+    if (defaultPagerDutyClient.isEmpty()) {
+      throw new RuntimeException("Default pager duty client is required");
```

Comment:
I believe we should unpack the PD client from the service when it is needed as others may not have PD setup which throw this exception. 

If the PD client isn't defined or it throws an exception (couldn't reach the PD servers) then increment a counter using the `meterRegistry`. We will be able to trigger an alert off of this counter going over a threshold. Also you can put the spiffe in the tag of the counter so it can be easily filtered.

Reply:
Excellent idea. I forgot about the meter registry for tracking this